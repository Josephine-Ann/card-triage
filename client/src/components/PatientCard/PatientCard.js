import React, { useContext } from 'react'
// import { usePatientInfo, usePatientInfoUpdate } from './PatientInfoProvider'
import { PatientInfoContext } from '../../PatientInfoProvider'
import './PatientCard.css'

export const PatientCard = (props) => {
    const {
        updatePatient,
        state: {
            done,
            pendingOrRejected
        }
    } = useContext(PatientInfoContext);
    return (
        <div className="card">
            <div id="card__info">
                <div>
                    <p className="card__para card__titles">General Information</p>
                    <p className="card__para ">Name:</p>
                    <p className="card__para ">{props.patient_name}</p>
                    <p className="card__para ">Reading taken on: </p>
                    <p className="card__para ">{props.created_date}</p>
                    <p className="card__para ">Status: {props.status[0]}{props.status.slice(1, 8).toLowerCase()}</p>
                    <p className="card__para  card__titles">Arrythmias</p>
                    <ul className="card__ul">
                        {props.arrhythmias.map((arrhythmia) => {
                            return <li key={props.arrhythmias.indexOf(arrhythmia)}>{arrhythmia}</li>
                        })}
                    </ul>
                </div>
                <div id="card__button-change">
                    <button className="card__button" id="card__change" onClick={() => updatePatient(props.id)}>{props.status === 'DONE' ? done : pendingOrRejected}</button>
                </div>
            </div>
        </div>
    )
}



export default PatientCard

