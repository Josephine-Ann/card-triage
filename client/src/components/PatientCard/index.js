export { PatientCard } from './PatientCard.js'