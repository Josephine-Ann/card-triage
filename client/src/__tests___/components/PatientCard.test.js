import React from 'react'
import { shallow } from 'enzyme';
import { PatientCard } from '../../components/PatientCard/PatientCard';
import patients from '../fixtures/patients';

jest.mock("react", () => ({
    ...jest.requireActual("react"),
    useContext: () => ({
        updateFilter: jest.fn()
    }),
}));



describe("PatientCard", () => {
    it("renders as expected in snapshot", () => {
        const wrapper = shallow(<PatientCard key={patients[0].id} {...patients[0]} />);
        expect(wrapper).toMatchSnapshot();
    });
});