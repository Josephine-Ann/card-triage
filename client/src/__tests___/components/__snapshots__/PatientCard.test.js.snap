// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientCard renders as expected in snapshot 1`] = `
ShallowWrapper {
  "length": 1,
}
`;
